elastic/kibana
# [7.x] Conditionally sets ignore_throttled only when search:includeFrozen is true

Fixes #113197

## Summary

[The "ignore_throttled" ES option is deprecated](https://github.com/elastic/elasticsearch/pull/77479), [as are frozen indices in general.](https://www.elastic.co/guide/en/elasticsearch/reference/current/frozen-indices.html). ES queries behave as though this flag is set to `true` by default, meaning they will ignore frozen indices. 

When the Kibana setting for "include frozen indices" is set to true, we still need to set this flag to `false` explicitly to continue to respect that Kibana setting. ([Exact deprecation strategy for that Kibana setting is still TBD.](https://github.com/elastic/kibana/issues/113197#issuecomment-938106133))

However, the current code was _always_ setting the `ignore_throttled` option to be the `!`-negated value of whatever the `search:includeFrozen` setting was. Whenever this value is included in a query, an ES deprecation log is produced. When a user hasn't indicated that they are even _trying_ to interact with the deprecated frozen index feature, we shouldn't be including this flag and setting it to its default value. We should leave the flag out altogether and avoid the deprecation log warning.

## To Test / Reproduce

1. Spin up this branch which should run Kibana in 7.16
2. Make sure the "include frozen" setting in "Advanced Settings" is off:

<img width="752" alt="Screen Shot 2021-10-07 at 10 44 46 PM" src="https://user-images.githubusercontent.com/159370/136490303-fc6795e6-73a6-4309-ada0-6a0d7e0dbf50.png">

3. Visit the Upgrade Assistant and turn on deprecation logging:

<img width="1428" alt="Screen Shot 2021-10-07 at 9 46 30 PM" src="https://user-images.githubusercontent.com/159370/136490503-950846f5-0c31-4af7-84f1-5f35295fad07.png">

4. Click "View deprecation logs in Observability" (Note: it will open the Logs UI with `?sourceId=deprecation_logs`, if you visit the Logs UI directly, you won't see them)
5. Live stream the logs
6. You shouldn't see any deprecation logs about "ignore_throttled"
7. Go back to "Advanced Settings" and turn on the "include frozen" setting
8. Return to the Upgrade Assistant and click on "View deprecation logs in Observability"
9. Live stream the logs
10. You should see deprecation logs about "ignore_throttled"

<img width="1087" alt="Screen Shot 2021-10-07 at 8 48 07 PM" src="https://user-images.githubusercontent.com/159370/136491719-25a92310-d091-4a6f-a818-21a75f2ae6fd.png">
